frontend/src/components/phones/dashboard/SendersPanelView.tsx

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable @typescript-eslint/no-non-null-asserted-optional-chain */
+import Moment from "react-moment";
+import styles from "./PhoneDashboard.module.scss";
+import {
+  ChevronLeftIcon,
+  ForwardedCallIcon,
+  ForwardedTextIcon,
+  WarningFilledIcon,
+} from "../../../components/Icons";
+import disabledSendersDataIllustration from "./images/sender-data-disabled-illustration.svg";
+import emptySenderDataIllustration from "./images/sender-data-empty-illustration.svg";
+import { useLocalization } from "@fluent/react";
+import { useInboundContact } from "../../../hooks/api/inboundContact";
+import moment from "moment";
+import { OutboundLink } from "react-ga";
+
+export type Props = {
+  type: "primary" | "disabled" | "empty";
+  back_btn: any;
+};
+
+export const SendersPanelView = (props: Props) => {
+  const { l10n } = useLocalization();
+  const inboundContactData = useInboundContact();
+  const inboundArray = inboundContactData.data;
+
+  const emptyCallerSMSSendersPanel = (
+    <div className={styles["senders-panel"]}>
+      <img
+        src={emptySenderDataIllustration.src}
+        alt="Empty Senders Data Illustration"
+        width={130}
+      />
+      <p className={styles["senders-panel-body"]}>
+        {l10n.getString("phone-dashboard-sender-empty-body")}
+      </p>
+    </div>
+  );
+
+  const disabledCallerSMSSendersPanel = (
+    <div className={styles["senders-panel"]}>
+      <img
+        src={disabledSendersDataIllustration.src}
+        alt="Disabled Senders Data Illustration"
+        width={130}
+      />
+      <p className={styles["senders-panel-body"]}>
+        <WarningFilledIcon
+          alt=""
+          className={styles["warning-icon"]}
+          width={20}
+          height={20}
+        />
+        {l10n.getString("phone-dashboard-sender-disabled-body")}
+      </p>
+      <div className={styles["update-settings-cta"]}>
+        <OutboundLink
+          to="/accounts/settings/"
+          eventLabel="Update Settings"
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          {l10n.getString("phone-dashboard-sender-disabled-update-settings")}
+        </OutboundLink>
+      </div>
+    </div>
+  );
+
+  const calendarStrings = {
+    lastDay: "[Yesterday at] LT",
+    sameDay: "[Today at] LT",
+    lastWeek: "L LT",
+    nextWeek: "L LT",
+    sameElse: "L LT",
+  };
+
+  const inboundContactArray =
+    inboundContactData &&
+    inboundArray
+      ?.sort(
+        (a, b) =>
+          // Sort by last sent date
+          moment(b.last_inbound_date).unix() -
+          moment(a.last_inbound_date).unix()
+      )
+      .map((data) => {
+        return (
+          <li
+            key={data.id}
+            className={data.blocked ? styles["greyed-contact"] : ""}
+          >
+            <span className={styles["sender-number"]}>
+              {formatPhoneNumberToUSDisplay(data.inbound_number)}
```

Comment:
change (blocking): use the new library to format phone numbers.

Reply:
Done (sad that u don't like my brilliant function)